### PR TITLE
chapter 15 created

### PR DESCRIPTION
**Pull request by agustinaGallo97**
## :link: Jira Tracker reference:
* [AP - Chapter 15] Implicit intents -  https://androidtraining.atlassian.net/browse/AT-102

## Description:
Chapter 15 created

## :camera_flash: UI preview
ezgif com-video-to-gif(15)

### CODE REVIEW

File: app/src/main/java/com/bignerdranch/android/criminalintent/CrimeRepository.kt [hunk 1]
```diff
@@ -23,7 +24,9 @@ class CrimeRepository private constructor() {
   }
 
   private val database: CrimeDatabase =
-    Room.databaseBuilder(CriminalIntentApplication.context, CrimeDatabase::class.java, DATABASE_NAME).build()
+    Room.databaseBuilder(CriminalIntentApplication.context, CrimeDatabase::class.java, DATABASE_NAME).addMigrations(
```
Comment (f7deleon) [inline app/src/main/java/com/bignerdranch/android/criminalintent/CrimeRepository.kt]: maybe `.addMigrations` in the next line?

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt [hunk 2]
```diff
@@ -68,6 +76,15 @@ class CrimeFragment : Fragment(R.layout.fragment_crime), DatePickerFragment.Call
     }
   }
 
+  private fun setUpVars(view: View) {
+    titleField = view.findViewById(R.id.crimeTitle) as EditText
+    dateButton = view.findViewById(R.id.crimeDate) as Button
+    timeButton = view.findViewById(R.id.crimeTime)
```
Comment (f7deleon) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt]: shouldn't this be casted as the others?

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt [hunk 3]
```diff
@@ -96,6 +113,27 @@ class CrimeFragment : Fragment(R.layout.fragment_crime), DatePickerFragment.Call
         show(this@CrimeFragment.requireFragmentManager(), DIALOG_TIME)
       }
     }
+
+    reportButton.setOnClickListener {
+      Intent(Intent.ACTION_SEND).apply {
+        type = "text/plain"
```
Comment (f7deleon) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt]: use constants for this

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -22,4 +22,7 @@
         </activity>
     </application>
 
+    <uses-permission android:name="android.permission.READ_CONTACTS" />
+    <uses-permission android:name="android.permission.CALL_PHONE" />
+
 </manifest>
```
Comment (f7deleon) [inline app/src/main/AndroidManifest.xml]: new line

**Review by f7deleon — APPROVED** [review]

File: app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -22,4 +22,7 @@
         </activity>
     </application>
 
+    <uses-permission android:name="android.permission.READ_CONTACTS" />
```
Comment (mirland) [inline app/src/main/AndroidManifest.xml:25]: we try to add at it the beginning of the file (above of application tag)

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt [hunk 6]
```diff
@@ -138,9 +139,11 @@ class CrimeFragment : Fragment(R.layout.fragment_crime), DatePickerFragment.Call
       if (resolvedActivity == null) isEnabled = false
     }
 
-//    callSuspectButton.apply {
-//      val callContactIntent = Intent(Intent.ACTION_DIAL, ContactsContract.CommonDataKinds.Phone)
-//    }
+    callSuspectButton.setOnClickListener {
+      val intent = Intent(Intent.ACTION_PICK)
+      intent.type = ContactsContract.CommonDataKinds.Phone.CONTENT_TYPE
+      startActivityForResult(intent, 1)
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt:145]: magic number

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt [hunk 7]
```diff
@@ -185,16 +188,32 @@ class CrimeFragment : Fragment(R.layout.fragment_crime), DatePickerFragment.Call
...
+        )
+        val cursor = requireActivity().contentResolver.query(
+          contactUri!!, queryFields, null, null, null
+        )
         cursor?.use {
-          if (it.count == 0) return
           it.moveToFirst()
-          val suspect = it.getString(0)
-          crime.suspect = suspect
+          var numberColumn = it.getColumnIndex(ContactsContract.CommonDataKinds.Phone.NUMBER)
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt:202]: the fragment shouldn't use a cursor! Use a repository

**Pull request merged**